src/python/WMCore/WorkerThreads/BaseWorkerThread.py

```diff
@@ -121,6 +144,9 @@ def initInThread(self, parameters):
         logging.info("Initialising default transaction")
         myThread.transaction = Transaction(myThread.dbi)
 
+        self.setUpHeartbeat()
+        self.setUpLogDB()
```

Comment:
I'm just not sure whether at this stage the worker threads setup has been done already(?)

Reply:
Worker thread set up is not done this stage. However, information necessary for heartbeat and logDB setting will be available at this point.  
Actually the error happened in worker node setup. So the heartbeat register didn't happen as you explained before.